pkgs/by-name/g-/g-golf-adw-1-examples/package.nix

```
@@ -0,0 +1,73 @@
+{
+  stdenv,
+  lib,
+  fetchurl,
+  makeWrapper,
+  wrapGAppsHook4,
+  pkg-config,
+  guile,
+  guile-cairo,
+  glib,
+  gobject-introspection,
+  libadwaita,
+  callPackage,
+  g-golf,
+  texinfo,
+  ...
+}:
+stdenv.mkDerivation (finalAttrs: {
+
+  pname = "g-golf-adw-1-examples";
+  version = "0.8.1";
+
+  src = fetchurl {
+    url = "http://ftp.gnu.org/gnu/g-golf/g-golf-${finalAttrs.version}.tar.gz";
+    hash = "sha256-yl5pSENYcOqx7TUWmzSXHhj02HuCr+ak4UsTtDIxjyI=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    texinfo
+    makeWrapper
+    wrapGAppsHook4
+  ];
+
+  buildInputs = [
+    guile
+    g-golf
+    guile-cairo
+    glib.out
+    gobject-introspection
+    libadwaita
+  ];
+
+  buildPhase = ''
+    pushd . && cd ./examples/adw-1/demo/
+    ${glib.dev}/bin/glib-compile-resources --target g-resources g-resources.xml
+    popd
+  '';
```

```suggestion
  sourceRoot = "${src}/examples/adw-1/demo";

  buildPhase = ''
    glib-compile-resources --target g-resources g-resources.xml
  '';
```
https://nixos.org/manual/nixpkgs/unstable/#var-stdenv-sourceRoot

Also add `glib` (without specifying output) to `nativeBuildInputs`